test/base/strict-mode.js

```diff
@@ -241,6 +244,147 @@ test("warn on unsafe reads", function() {
     }
 })
 
+describe("observableRequiresReaction", function() {
+    test("warn on unsafe reads of observable", function() {
+        try {
+            mobx.configure({ observableRequiresReaction: true })
+            const x = mobx.observable({
+                y: 3
+            })
+            utils.consoleWarn(() => {
+                x.yy
+            }, /being read outside a reactive context/)
+        } finally {
+            mobx.configure({ observableRequiresReaction: false })
+        }
+    })
+
+    test("warn on unsafe reads of observable also when there are other subscriptions", function() {
+        try {
+            mobx.configure({ observableRequiresReaction: true })
+            const x = mobx.observable({
+                y: 3
+            })
+
+            const dispose = mobx.autorun(() => x.y)
+
+            utils.consoleWarn(() => {
+                x.y
+            }, /being read outside a reactive context/)
+
+            dispose()
+        } finally {
+            mobx.configure({ observableRequiresReaction: false })
+        }
+    })
+
+    test("warn on unsafe reads of observable array", function() {
+        try {
+            mobx.configure({ observableRequiresReaction: true })
+            const x = mobx.observable({
+                arr: [1, 2, 3]
+            })
+            utils.consoleWarn(() => {
+                x.arr[1]
+            }, /being read outside a reactive context/)
+        } finally {
+            mobx.configure({ observableRequiresReaction: false })
+        }
+    })
+    test("don't warn on reads inside batch - computed", function() {
```

Comment:
`inside batch - computed` => `inside computed` (not a batch)

Reply:
its also inside a batch,
That why its not warning